zilliqa/src/cfg.rs

```diff
@@ -99,7 +102,7 @@ impl Default for ConsensusConfig {
             rewards_per_hour: 32_000_000_000_000u128,
             blocks_per_hour: 50_000,
             minimum_stake: 32_000_000_000_000_000_000u128,
-            block_gas_limit: 84_000_000,
+            eth_block_gas_limit: EvmGas(84_000_000),
             gas_price: 4_761_904_800_000,
```

Comment:
Small nit - I think we should remove these defaults or remove the values from `infra/config.toml` - I don't like having the 'sensible' values in two places where they can diverge. My light preference would be that we remove these defaults and force people to have to specify some values in `config.toml`?

Reply:
Maybe it makes more sense to remove the `Default` impl for `Config` and instead just default the fields that do have sensible values with `#[serde(default = "...")]`? I agree with you that having `Option`s here and unwrapping everywhere is quite horrible.

Reply:
I think the bigger problem with making them optional and unwrapping everywhere is that the error messages to node operators who forget to fill these fields will be much worse. Instead of "missing field X" at startup, their node will start but then later they will get a random panic and have to search the backtrace to work out why.

How painful would removing the `Default` impl be? If it is painful, then I guess we can merge this for now and change it later? Up to you.

Reply:
Thanks :)